deluge/common.py

```diff
@@ -46,6 +46,9 @@
     from urlparse import urljoin  # pylint: disable=ungrouped-imports
     from urllib import pathname2url, unquote_plus  # pylint: disable=ungrouped-imports
 
+if platform.system() == 'Linux':
+    import distro
```

Comment:
Since the distro details are not that important this would be better as optional package

Reply:
How do I do that? Isn't it by stating in the requires section in `setup.py` (which I forgot to add)?

Reply:
This will importerror if distro is not installed, have a look at dbus below for example

Reply:
Oh, OK.
But now I got a question... Should I add `distro` to `requirements.txt` (or any other file of them) and to `install_requires` in `setup.py`?

Reply:
> Should I add `distro` to `requirements.txt` (or any other file of them) and to `install_requires` in `setup.py`?

Yes add to both. `install_requires` list is automatically installed by pip and requirements.txt is for manual environment setup with `pip -r`. 

I have put optional packages in `install_requires` because they are soft-required for best user-experience and would be a headache to put in `extra_require` and ensure they are installed. 

My reasoning for making `distro` package optional is to help with packaging. For example there is no deb package on Ubuntu so Deluge would break if it's missing.

Reply:
Oh don't forget to update the DEPENDS file too.